src/toast/Toast.tsx

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as React from "react";
+import "@clr/icons/clr-icons.min.css";
+import "@webcomponents/custom-elements/custom-elements.min.js";
+import "@clr/icons/clr-icons.min.js";
+import {Icon} from "../icon";
+import "./styles.css";
+
+export enum MessageType {
+    SUCCESS = "success",
+    WARNING = "warning",
+    DANGER = "danger",
+    INFO = "info",
+}
+
+export enum Direction {
+    TOP_RIGHT = "top-right",
+    BOTTOM_LEFT = "bottom-left",
+    BOTTOM_RIGHT = "bottom-right",
+    TOP_LEFT = "top-left",
+    LEFT = "left",
+    RIGHT = "right",
+    BOTTOM = "bottom",
+    TOP = "top",
+    CAPS_LEFT = "Left",
+    CAPS_RIGHT = "Right",
+    CAPS_BOTTOM = "Bottom",
+    CAPS_TOP = "Top",
+}
+
+const MARGINS = {
+    POS_MARGIN: "15px",
+    NEG_MARGIN: "-15px",
+};
+
+const ICONTYPE = {
+    WARNING: "dell-alert-warning",
+    SUCCESS: "success-standard",
+    DANGER: "error-standard",
+    INFO: "dell-alert-info",
+};
+
+interface ToastProps {
+    notify: (fn: Function) => void;
+}
+export default class Toast extends React.Component<ToastProps> {
+    constructor(props: ToastProps) {
+        super(props);
+        props.displayToast(this.show_toast);
+
+        this.state = {
```

Comment:
you can keep this outside of constructor.

Reply:
Addressed this comment